### PR TITLE
test(state): removed guards, added tests

### PR DESCRIPTION
**Pull request by BioPhoton**

### CODE REVIEW

File: libs/state/spec/core/utils/pipe-from-array.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+describe('pipeFromArray', () => {
+
+  it('should return true for arrays of function', () => {
+    expect(pipeFromArray(undefined as any)).toBeTruthy();
+    expect(pipeFromArray([])).toBeTruthy();
+    expect(pipeFromArray([map(() => {})])).toBeTruthy();
+    expect(pipeFromArray([map(() => {}), () => {}])).toBeTruthy();
+  });
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/pipe-from-array.spec.ts:19]: why don't we test that the result is type of function?

File: libs/state/spec/core/utils/typing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  it('should return true for a Promise', () => {
+    expect(isPromiseGuard(new Promise(() => {
+    }))).toBeTruthy();
+    expect(isPromiseGuard({
+      then: () => {
+      }
+    })).toBeTruthy();
+  });
+
+  it('should return false for no Promise', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:21]:
```suggestion
  it('should return false if input is not a promise', () => {
```

File: libs/state/spec/core/utils/typing.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    expect(isPromiseGuard(true)).toBeFalsy();
+    expect(isPromiseGuard({ then: true })).toBeFalsy();
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:23]: some more test cases?

File: libs/state/spec/core/utils/pipe-from-array.spec.ts (hunk 1 above)
Comment (BioPhoton) [reply libs/state/spec/core/utils/pipe-from-array.spec.ts]: done

File: libs/state/spec/core/utils/typing.spec.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    expect(isOperateFnArrayGuard([] as any)).toBeFalsy();
+    expect(isOperateFnArrayGuard([[], [], []])).toBeFalsy();
+    expect(isOperateFnArrayGuard([of(1), from([]), of(2)])).toBeFalsy();
+    expect(isOperateFnArrayGuard(of(1) as any)).toBeFalsy();
+  });
+});
+
+describe('isStringArrayGuard', () => {
+
+  it('should return true for a array of strings', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:53]:
```suggestion
  it('should return true for an array of strings', () => {
```

File: libs/state/spec/core/utils/typing.spec.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    expect(isStringArrayGuard(['1', '2', '3'])).toBeTruthy();
+  });
+
+  it('should return false for other values', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:57]:
```suggestion
  it('should return false for other input types', () => {
```

File: libs/state/spec/core/utils/typing.spec.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    expect(isStringArrayGuard([{}, {}, {}])).toBeFalsy();
+    expect(isStringArrayGuard([] as any)).toBeFalsy();
+    expect(isStringArrayGuard([[], [], []])).toBeFalsy();
+    expect(isStringArrayGuard([of(1), from([]), of(2)])).toBeFalsy();
+    expect(isStringArrayGuard(of(1) as any)).toBeFalsy();
+  });
+});
+
+describe('isIterableGuard', () => {
+  it('should return true for a Promise', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:74]:
```suggestion
  it('should return true if input is a Promise', () => {
```

File: libs/state/spec/core/utils/typing.spec.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    expect(isIterableGuard([])).toBeTruthy();
+    expect(isIterableGuard({[Symbol.iterator]: () => {}})).toBeTruthy();
+  });
+
+  it('should return false for no Promise', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:79]:
```suggestion
  it('should return false for input types other than Promise', () => {
```

File: libs/state/spec/core/utils/typing.spec.ts [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+});
+
+describe('isKeyOf', () => {
+  it('should return true if key exists', () => {
+    expect(isKeyOf<{num: number}>('num')).toBeTruthy();
+    expect(isKeyOf<{1: number}>(1)).toBeTruthy();
+    expect(isKeyOf<{[Symbol.iterator]: number}>(Symbol.iterator)).toBeTruthy();
+  });
+
+  it('should return false for no Promise', () => {
```
Comment (hoebbelsB) [inline libs/state/spec/core/utils/typing.spec.ts:94]: ????

File: libs/state/spec/core/utils/typing.spec.ts (hunk 3 above)
Comment (BioPhoton) [reply libs/state/spec/core/utils/typing.spec.ts]: done

File: libs/state/spec/core/utils/typing.spec.ts (hunk 8 above)
Comment (BioPhoton) [reply libs/state/spec/core/utils/typing.spec.ts:94]: uups this was a copy-paste bug :D

**Pull request merged**